src/cargo/ops/cargo_package.rs

```diff
@@ -35,7 +35,6 @@ pub struct PackageOpts<'cfg> {
 static VCS_INFO_FILE: &'static str = ".cargo_vcs_info.json";
 
 pub fn package(ws: &Workspace<'_>, opts: &PackageOpts<'_>) -> CargoResult<Option<FileLock>> {
-    ops::resolve_ws(ws)?;
```

Comment:
We may want to keep doing this in the sense that it ensures that the package doesn't have stray deps that go to the middle of nowhere and things like that. I'd be somewhat wary of removing this in the sense that you may be able to publish where manifests are messed up in a situation where you publish a valid sub-crate from a workspace that is otherwise currently corrupt.

I do agree that this is mostly opportunistic at that point in the sense that it's quickly discarded, but it seems like good hygiene for publication no matter what.

Reply:
Do you have any ideas on how to avoid the index from being updated multiple times?  If I add this back, it would get updated 3 times (once here, once for constructing the new Cargo.lock, and once during verify).

Reply:
Hm a good point. We ran into this awhile back and it's why functions like `resolve_ws_precisely` take a `source` option (something that's already preloaded). The general idea is that the source needs to be preloaded into the `Registry` somehow (the `cargo::core::Registry`). 

We could try to thread through some sources, but it's honestly probably easier to keep a `HashSet<SourceId>` of updated registries in `Config` and basically just avoid updating if it's already in there.